utils_cv/detection/plot.py

```diff
@@ -116,6 +131,98 @@ def display_bounding_boxes(
         plt.show()
 
 
+def plot_grid(
+    plot_func: Callable[..., None],
+    args: Union[Callable, Iterator, Any],
+    rows: int = 1,
+    cols: int = 3,
+    figsize: Tuple[int, int] = (16, 16),
+) -> None:
+    """ Helper function to plot image grids.
+
+    Args:
+        plot_func: callback to call on each subplot. It should take an 'ax' as
+        the last param.
+        args: args can be passed in in many forms. It can be an iterator, a
+        callable, or simply some static parameters. If it is an iterator, this
+        function will call `next` on it each time. If it is a callable, this
+        function will call the function and use the returned values each time.
+        rows: rows to plot
+        cols: cols to plot, default is 3. NOTE: use cols=3 for best looking
+        grid
+        figsize: figure size (will be dynamically modified in the code
+
+    Returns nothing but plots graph
+    """
+    fig_height = rows * 8
+    figsize = (16, fig_height)
+
+    fig, axes = plt.subplots(rows, cols, figsize=figsize)
+
+    if rows == 1:
+        axes = [axes]
+
+    for row in axes:
+        for ax in row:
+            # dynamic injection of params into callable
+            arguments = (
+                args()
+                if isinstance(args, Callable)
+                else (next(args) if hasattr(args, "__iter__") else args)
+            )
+            try:
+                plot_func(arguments, ax)
+            except Exception:
+                plot_func(*arguments, ax)
+
+    plt.subplots_adjust(top=0.8, bottom=0.2, hspace=0.1, wspace=0.2)
+
+
+def plot_detection_vs_ground_truth(
+    im_path: str,
+    detection_annos: List[AnnotationBbox],
+    ground_truth_annos: List[AnnotationBbox],
+    ax: plt.axes,
+) -> None:
+    """ Plots bounding boxes of ground_truths and detections.
+
+    Args:
+        im_path: the image to plot
+        detection_annos: a list of detected annotations
+        ground_truth_annos: a list of ground_truth detections
+        ax: the axis to plot on
+
+    Returns nothing, but displays a graph
+    """
+    im = Image.open(im_path).convert("RGB")
+
+    # plot detections
+    pred_params = PlotSettings(rect_color=(255, 0, 0), text_size=1)
+    im = plot_boxes(
+        im,
+        detection_annos,
+        im_title=os.path.basename(im_path),
+        plot_settings=pred_params,
+    )
+
+    # plot ground truth boxes
+    ground_truth_params = PlotSettings(rect_color=(0, 255, 0), text_size=1)
+    im = plot_boxes(
+        im,
+        ground_truth_annos,
+        im_title=os.path.basename(im_path),
+        plot_settings=ground_truth_params,
+    )
+
+    # show image
+    ax.set_xticks([])
+    ax.set_yticks([])
+    ax.imshow(im)
+
+
+# ===== Precision - Recall curve =====
+
+
 def _get_cmap(n, name="hsv"):
```

Comment:
At the moment this function is only used once, hence not sure the function is needed.